src/js/components/DataFilter/DataFilter.js

```diff
@@ -0,0 +1,177 @@
+
+    // generate options from all values for property
+    const uniqueValues = generateOptions(unfilteredData, property);
+    // if any values aren't numeric, treat as options
+    if (uniqueValues.some((v) => v && typeof v !== 'number'))
+      return uniqueValues;
+    // if all values are numeric, let range take care of it
+    return undefined;
+  }, [children, optionsProp, properties, property, rangeProp, unfilteredData]);
+
+  const range = useMemo(() => {
+    if (children) return undefined; // caller driving
+    if (rangeProp) return rangeProp; // caller setting
+    // Data properties setting
+    if (properties?.[property]?.range) {
+      const { min, max } = properties[property].range;
+      return [min, max];
+    }
+    // skip if we have options
+    if (options) return undefined;
+
+    // generate range from all values for the property
+    const uniqueValues = generateOptions(unfilteredData, property).sort();
+    // normalize to make it friendler, so [1.3, 4.895] becomes [1, 5]
+    const delta = uniqueValues[uniqueValues.length - 1] - uniqueValues[0];
+    const interval = Number.parseFloat((delta / 3).toPrecision(1));
+    const min = alignMin(uniqueValues[0], interval);
+    const max = alignMax(uniqueValues[uniqueValues.length - 1], interval);
+    return [min, max];
+  }, [children, options, properties, property, rangeProp, unfilteredData]);
+
+  // track range values so we can display them
+  const [rangeValues, setRangeValues] = useState(() => {
+    if (!range) return undefined;
+    const { min, max } = view?.properties?.[property] || {};
+    return [min || range[0], max || range[1]];
+  }, [range, view]);
+
+  // keep the text values size consistent
+  useLayoutEffect(() => {
+    if (maxRef.current && minRef.current) {
+      const width = Math.max(
+        maxRef.current.getBoundingClientRect().width,
+        minRef.current.getBoundingClientRect().width,
+      );
+      maxRef.current.style.width = `${width}px`;
+      minRef.current.style.width = `${width}px`;
+    }
+  });
+
+  const id = `${dataId}-${property}`;
+
+  let content = children;
+  if (!content) {
+    if (range) {
+      // consider moving the Text values inside RangeSelector
+      content = (
+        <Box
+          direction="row"
+          justify="between"
+          align="center"
+          pad="xsmall"
+          gap="small"
+        >
+          <Text ref={minRef} size="small" style={{ fontFamily: 'monospace' }}>
+            {rangeValues[0]}
+          </Text>
+          <RangeSelector
+            id={id}
+            name={`${property}._range`}
+            defaultValues={range}
+            direction="horizontal"
+            invert={false}
+            min={range[0]}
+            max={range[1]}
+            step={(range[1] - range[0]) / 20}
+            size="full"
+            round="small"
+            onChange={setRangeValues}
+          />
+          <Text ref={maxRef} size="small" style={{ fontFamily: 'monospace' }}>
```

Comment:
not a inline style! 😅 I like the choice though for text style 👍

Reply:
Removed use of monospace as it didn't seem to align with anything else we have elsewhere. I'm thinking we'd want to generally look at how we use that to be more consistent.